service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -170,23 +173,31 @@ public static void rethrowIfSamInterrupted(VoidInterruptedSupplier function, Str
       throw SamExceptionFactory.create("Interrupted during Sam operation " + operation, e);
     }
   }
+
   /**
    * Register WSM's service account as a user in Sam if it isn't already. This is called on every
    * WSM startup, but should only need to register with Sam once per environment.
    */
-  public void initialize() throws InterruptedException {
-    String wsmAccessToken;
-    try {
-      wsmAccessToken = getWsmServiceAccountToken();
-    } catch (IOException e) {
-      // In cases where WSM is not running as a service account (e.g. unit tests), the above call
-      // will throw. This can be ignored now and later when the credentials are used again.
-      logger.warn("Failed to register WSM service account in Sam. This is expected for tests.", e);
-      return;
-    }
-    UsersApi usersApi = samUsersApi(wsmAccessToken);
-    if (!wsmServiceAccountRegistered(usersApi)) {
-      registerWsmServiceAccount(usersApi);
+  private void initialize() throws InterruptedException {
+    if (!initialized) {
+      String wsmAccessToken = null;
+      try {
+        wsmAccessToken = getWsmServiceAccountToken();
+      } catch (IOException e) {
+        // In cases where WSM is not running as a service account (e.g. unit tests), the above call
+        // will throw. This can be ignored now and later when the credentials are used again.
+        logger.warn(
+            "Failed to register WSM service account in Sam. This is expected for tests.", e);
+        return;
+      }
+      UsersApi usersApi = samUsersApi(wsmAccessToken);
+      // If registering the service account fails, all we can do is to keep trying.
```

Comment:
I don't think we should wait forever. This is now happening due to a REST API request.  I'd suggest doing this operation using the register operation using the sam retry code that wraps other sam calls so it has the same failure behavior.